### PR TITLE
Update the matrix-sdk crates to 9b05d0d822

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

**Review by Hywan — APPROVED** [review]
LGTM, thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # UNRELEASED
 
+**BREAKING CHANGES**
+
+-   Update matrix-rust-sdk to `9b05d0d82`, which includes:
+
+    -   `Device.requestVerification`, `UserIdentities.requestVerification`, and
+        `UserIdentities.verificationRequestContent` is not async anymore.
+        ([#3513](https://github.com/matrix-org/matrix-rust-sdk/pull/3513))
```
Comment (richvdh) [inline CHANGELOG.md:9]: Worth mentioning that converting an async function to not-async is not normally a breaking change in JS (though typescript might complain at you about the redundant `await`).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
+    -   Use the server name in the `QrCodeData` instead of the homeserver URL.
+        ([#3537](https://github.com/matrix-org/matrix-rust-sdk/pull/3537))
```
Comment (richvdh) [inline CHANGELOG.md:12]: Not sure if this is a breaking change either?

File: CHANGELOG.md (hunk 1 above)
Comment (poljar) [reply CHANGELOG.md:9]: Oh that's unexpected, thanks for letting me know.

File: CHANGELOG.md (hunk 2 above)
Comment (poljar) [reply CHANGELOG.md:12]: Well we did rename the field, and the type of the field, thus the getter in the bindings was also renamed.

File: CHANGELOG.md (hunk 1 above)
Comment (t3chguy) [reply CHANGELOG.md:9]: > Worth mentioning that converting an async function to not-async is not normally a breaking change in JS (though typescript might complain at you about the redundant await).

I disagree, as the method return value loses `.then` and `.catch` properties for non-async-await envs.

File: CHANGELOG.md (hunk 2 above)
Comment (richvdh) [reply CHANGELOG.md:12]: ah, it might be good to call out that rename then?

Comment (poljar) [reply CHANGELOG.md:12]: https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/125

**Pull request merged**